qa/common/common.sh

```diff
@@ -577,3 +570,35 @@ echo "Result: OK"
 EOF
     _run_test_script_on_node $TESTSCRIPT $RGWNODE
 }
+
+function test_systemd_ceph_osd_target_wants {
+  #
+  # see bsc#1051598 in which ceph-disk was omitting --runtime when it enabled
+  # ceph-osd@$ID.service units
+  #
+  local CEPH_OSD_WANTS="/systemd/system/ceph-osd.target.wants"
+  local ETC_CEPH_OSD_WANTS="/etc$CEPH_OSD_WANTS"
+  local RUN_CEPH_OSD_WANTS="/run$CEPH_OSD_WANTS"
+  local DIR_CONTENTS
+  set +e
+  ls -l $ETC_CEPH_OSD_WANTS
+  ls -l $RUN_CEPH_OSD_WANTS
+  # if $ETC_... is not a directory and $RUN_... is, consider the test a pass
+  if [ -d "$RUN_CEPH_OSD_WANTS" -a ! -d "$ETC_CEPH_OSD_WANTS" ] ; then
+      echo "ceph-osd.target symlinks appear to be in the right place"
+      return 0
+  fi
+  set -e
+  # if $ETC_... is a directory, it should at least be empty
+  if [ -d "$ETC_CEPH_OSD_WANTS" ] ; then
+      DIR_CONTENTS=$(ls --almost-all $ETC_CEPH_OSD_WANTS)
+      echo "$ETC_CEPH_OSD_WANTS exists and is a directory: asserting that it is empty"
+      test -z "$DIR_CONTENTS"
```

Comment:
That directory is exceedingly unlikely to exist and be empty (in my experience it's removed after you `systemctl disable` the last osd). No harm allowing it to exist and be empty though for the purpose of this test.

Reply:
This if clause is there to catch the failure case. Right now after deploying SES5 the directory exists and is not empty, so the test (correctly) fails on this line.